go/api/commands.go

```diff
@@ -155,3 +155,228 @@ type StringCommands interface {
+//
+// See [valkey.io] for details.
+//
+// [valkey.io]: https://valkey.io/commands/?group=hash
+type HashCommands interface {
+	// HGet returns the value associated with field in the hash stored at key.
+	//
+	// See [valkey.io] for details.
+	//
+	// For example:
+	//
+	//    result, err := client.HGet("myhash", "field1")
+	//
+	// [valkey.io]: https://valkey.io/commands/hget/
+	HGet(key, field string) (string, error)
+
+	// HGetAll returns all fields and values of the hash stored at key.
+	//
+	// See [valkey.io] for details.
+	//
+	// Parameters:
+	//  key - The key of the hash to retrieve.
+	//
+	// Return value:
+	//  A map of all fields and their values in the hash, or an empty map when key does not exist.
+	//
+	// Complexity: O(N) where N is the size of the hash.
+	//
+	// Example:
+	//    result, err := client.HGetAll("myhash")
+	//    // result could be: map[string]string{"field1": "Hello", "field2": "World"}
+	//
+	// [valkey.io]: https://valkey.io/commands/hgetall/
+	HGetAll(key string) (map[string]string, error)
+
+	// HMGet returns the values associated with the specified fields in the hash stored at key.
+	//
+	// See [valkey.io] for details.
+	//
+	// For example:
+	//
+	//    result, err := client.HMGet("myhash", []string{"field1", "field2", "nofield"})
+	//
+	// [valkey.io]: https://valkey.io/commands/hmget/
+	HMGet(key string, fields []string) ([]string, error)
+
+	// HSet sets the specified fields to their respective values in the hash stored at key.
+	// This command overwrites the values of specified fields that exist in the hash.
+	// If key doesn't exist, a new key holding a hash is created.
+	//
+	// See [valkey.io] for details.
+	//
+	// Parameters:
+	//  key        - The key of the hash.
+	//  fields     - A map of field-value pairs to set in the map.
+	//
+	// Return value:
+	//  The number of fields that were added or updated.
+	//
+	// Complexity: O(1) for each field/value pair added, so O(N) to add N field/value pairs.
+	//
+	// Example:
+	//    fieldsToSet := map[string]string{
+	//        "field1": "Hello",
+	//        "field2": "Hi",
+	//        "field3": "World",
+	//    }
+	//    addedCount, err := client.HSet("myhash", fieldsToSet)
+	//    // addedCount will be the number of fields added or updated
+	//
+	// [valkey.io]: https://valkey.io/commands/hset/
+	HSet(key string, fields map[string]string) (int64, error)
+
+	// HSetNX sets field in the hash stored at key to value, only if field does not yet exist.
+	// If key does not exist, a new key holding a hash is created.
+	// If field already exists, this operation has no effect.
+	//
+	// See [valkey.io] for details.
+	//
+	// Parameters:
+	//  key   - The key of the hash.
+	//  field - The field to set.
+	//  value - The value to set.
+	//
+	// Return value:
+	//  1 if field is a new field in the hash and value was set.
+	//  0 if field already exists in the hash and no operation was performed.
```

Comment:
Returns true or false - we do value conversion. Don't trust redis/valkey docs, copy them from other clients